### PR TITLE
ddl: make added count correct in adding index

### LINKED ISSUE #6979: Make the value of "taskAddedCount" correct when adding index
If we retry in "backfillIndexInTxn" when we encounter an error, the adding count will wrong in adding index phase.

### PR DESCRIPTION
**Pull request by zimulala**
## What have you changed? (mandatory)

fix #6979
"RunInNewTxn" will do retry by itself when it encounters an error.

## What are the type of the changes (mandatory)?
- Improvement

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @tiancaiamao @coocood @ciscoxll @crazycs520

**Comment by crazycs520** [comment]
If backfillIndexInTxn encounter error, "RunInNewTxn" do retry will also backfill the index which was already filled before encounter error, right?

**Comment by zimulala** [comment]
@crazycs520 
In "backfillIndexInTxn" we only have the function of "RunInNewTxn". So if "backfillIndexInTxn" encounters an error, "RunInNewTxn" must encounter an error too. And if "RunInNewTxn" encounters an error, it will roll back by itself.

File: ddl/index.go [hunk 1]
```diff
@@ -587,9 +587,9 @@ func (w *addIndexWorker) fetchRowColVals(txn kv.Transaction, taskRange reorgInde
 // backfillIndexInTxn will add w.batchCnt indices once, default value of w.batchCnt is 128.
 // TODO: make w.batchCnt can be modified by system variable.
 func (w *addIndexWorker) backfillIndexInTxn(handleRange reorgIndexTask) (nextHandle int64, addedCount, scanCount int, errInTxn error) {
-	addedCount = 0
-	scanCount = 0
 	errInTxn = kv.RunInNewTxn(w.sessCtx.GetStore(), true, func(txn kv.Transaction) error {
+		addedCount = 0
+		scanCount = 0
 		txn.SetOption(kv.Priority, kv.PriorityLow)
 		idxRecords, handleOutOfRange, err := w.fetchRowColVals(txn, handleRange)
```
Comment (crazycs520) [inline ddl/index.go]: The indexRecord which was already filled before encounter error will also be backfilled again in retry process, right?

Comment (zimulala) [reply ddl/index.go]: @crazycs520 
Yes, it will be backfilled again. If the `RunInNewTxn` meets an error, these code lines from 593 to 625 need to retry.

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
There are 5 LGTMs in this pr. But no one wants to approve it. :)

**Comment by shenli** [comment]
@jackysp I made the third LGTM, but I am waiting for the CI.

**Pull request merged**